domain/secret/service/service.go

```diff
@@ -55,27 +63,64 @@ type SecretService struct {
 	adminConfigGetter BackendAdminConfigGetter
 }
 
+// CreateSecretURIs returns the specified number of new secret URIs.
 func (s *SecretService) CreateSecretURIs(ctx context.Context, count int) ([]*secrets.URI, error) {
 	if count <= 0 {
 		return nil, errors.NotValidf("secret URi count %d", count)
 	}
 
-	// TODO(secrets)
-	modelUUID := ""
+	modelUUID, err := s.st.GetModelUUID(ctx)
+	if err != nil {
+		return nil, errors.Annotate(err, "getting model uuid")
+	}
 	result := make([]*secrets.URI, count)
 	for i := 0; i < count; i++ {
 		result[i] = secrets.NewURI().WithSource(modelUUID)
 	}
 	return result, nil
 }
-func (s *SecretService) CreateSecret(ctx context.Context, uri *secrets.URI, params CreateSecretParams) (*secrets.SecretMetadata, error) {
-	return nil, errors.NotImplemented
-	/*
-		var nextRotateTime *time.Time
-		if params.RotatePolicy.WillRotate() {
-			nextRotateTime = params.RotatePolicy.NextRotateTime(s.clock.Now())
+
+// TODO(secrets) - we need to tweak the upsert params to avoid this
```

Comment:
Do we still wanna leave it as TODO?

Reply:
Yes, because the change will be done in a follow up PR when the update functionality is added. This PR is just about create.